src/ro/sci/h9/SkiRace.java

```diff
@@ -0,0 +1,63 @@
+package ro.sci.h9;
+
+import java.util.Comparator;
+import java.util.Set;
+import java.util.TreeSet;
+
+/**
+ * This class represents a ski race.
+ *
+ * @author Emanuel Pruker
+ */
+public class SkiRace {
+    private static String[] top3 = {"Winner", "Runner-up", "Third Place"};
+    private Set<Result> results;
+
+    /**
+     * Constructs an instance from the contents of a csv file.
+     *
+     * @param input the input string
+     */
+    public SkiRace(String input) {
+        if (input == null) {
+            throw new IllegalArgumentException("The input cannot be null");
+        }
+        results = new TreeSet<>(new ResultComparator());
+        for (String line : input.split("\n")) {
+            results.add(Result.fromString(line));
+        }
+    }
+
+    /**
+     * Constructs an instance from a list of results.
+     *
+     * @param results the list of results
+     */
+    public SkiRace(Set<Result> results) {
+        this.results = results;
+    }
+
+    /**
+     * Calculates the top 3 results of the race.
+     *
+     * @return the calculated result as a string
+     */
+    public String printTop3() {
+        StringBuilder builder = new StringBuilder();
+        Object[] resultsArray = results.toArray();
```

Comment:
not really needed because you create another object here but sets are iterable. Could have used results.iterator()
or java8 streams.
You basically build an array that has the same elements as the set. What if the set is really large? You double the amount of memory the set takes and you only use the first three elements